instrumentation/spring/spring-webflux/spring-webflux-5.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webflux/v5_0/server/RouterFunctionInstrumentation.java

```diff
@@ -61,11 +65,31 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class RouteAdvice {
 
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void methodEnter(
+        @Advice.Argument(0) ServerRequest request,
+        @Advice.Local("otelContext") Context context,
+        @Advice.Local("otelScope") Scope scope) {
+      Context parentContext = Context.current();
+      if (!instrumenter().shouldStart(parentContext, request)) {
+        return;
+      }
+
+      context = instrumenter().start(parentContext, request);
+      scope = context.makeCurrent();
+    }
+
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void methodExit(
         @Advice.This RouterFunction<?> thiz,
         @Advice.Return(readOnly = false) Mono<HandlerFunction<?>> result,
-        @Advice.Thrown Throwable throwable) {
+        @Advice.Thrown Throwable throwable,
+        @Advice.Local("otelContext") Context context,
```

Comment:
`context` is not used anywhere within this method; the span started in the `OnMethodEnter` advice does not seem to be ended. Is that correct? Is this span needed?

Reply:
span is needed for initializing the ServletContextPath. I have updated the code to end advice.

Reply:
You shouldn't need to create a span to use `ServletContextPath`. The `ServletContextPath` is supposed to be initialized somewhere in the application server layer, in a different instrumentation; here you're just reading whatever value was set before.